ui/src/app/utils/runtime-presets.ts

```diff
@@ -70,7 +67,19 @@ export const applyPresetOverride = (runtime) => {
     const { gceConfig, gceWithPdConfig, dataprocConfig } =
       runtimePresets[runtimePresetKey].runtimeTemplate;
 
-    return { ...runtime, gceConfig, gceWithPdConfig, dataprocConfig };
+    return {
+      ...runtime,
+      gceConfig,
+      // restore the original PD name, which will cause a creation request to attach it to the new runtime
```

Comment:
This is the other part of the fix: don't overwrite the PD name, if one exists

Reply:
Great comment and great PR comment.